root/static/scripts/edit/externalLinks.js

```diff
@@ -233,25 +233,26 @@ export class ExternalLinksEditor
             const oldLink = oldLinks[link.relationship];
             const isNewLink = !isPositiveInteger(link.relationship);
             const linkChanged = oldLink && link.url !== oldLink.url;
+            const isNewOrChangedLink = (isNewLink || linkChanged);
             const linkTypeChanged = oldLink && +link.type !== +oldLink.type;
             link.url = getUnicodeUrl(link.url);
 
             if (isEmpty(link)) {
               error = '';
             } else if (!link.url) {
               error = l('Required field.');
-            } else if (!isValidURL(link.url)) {
+            } else if (isNewOrChangedLink && !isValidURL(link.url)) {
```

Comment:
The logic looks correct, so approving, though I feel like we should have a nested `if` for all these checks that depend on `isNewOrChangedLink` instead of checking it 6 times:

```suggestion
            } else if (isNewOrChangedLink) {
              if (!isValidURL(link.url)) {
              } else if (...) {
                // ...
              }
            }
```

Maybe even move all the checks the depend only on the URL to a separate function. These checks were already messy before this PR so it'd be nice to make it a bit better.

Reply:
Let's merge for now, but this should be improved in a later PR, maybe make it support an array of errors.

Reply:
Added https://tickets.metabrainz.org/browse/MBS-11521